### PR TITLE
fix: Mention a user when changing the audience - MEED-2463 - Meeds-io/MIP#54 

### PR DESCRIPTION
**Pull request by MayTekayaa**
This change will mention a user according to the selected audience.

### CODE REVIEW

File: component/core/src/main/java/org/exoplatform/social/core/utils/MentionUtils.java [hunk 1]
```diff
@@ -18,7 +20,7 @@ public class MentionUtils {
 
     private static final Log LOG                             = ExoLogger.getLogger(MentionUtils.class);
 
-    public static String substituteUsernames(String portalOwner, String message) {
+    public static String substituteUsernames(String portalOwner, String message, ExoSocialActivity activity) {
```
Comment (boubaker) [inline component/core/src/main/java/org/exoplatform/social/core/utils/MentionUtils.java]: Keep method with two parameters available to not add a breaking change for upper addons.

```suggestion
    public static String substituteUsernames(String portalOwner, String message) {
      substituteUsernames(portalOwner, message);
    }

    public static String substituteUsernames(String portalOwner, String message, ExoSocialActivity activity) {
```

File: component/core/src/test/java/org/exoplatform/social/core/utils/MentionUtilsTest.java [hunk 2]
```diff
@@ -33,15 +33,15 @@ public void testSubstituteUsernames(){
         String portalOwner = "dw";
         String message = "hello <p>@root</p>";
 
-        message = MentionUtils.substituteUsernames(portalOwner, message);
+        message = MentionUtils.substituteUsernames(portalOwner, message, null);
```
Comment (boubaker) [inline component/core/src/test/java/org/exoplatform/social/core/utils/MentionUtilsTest.java]:
```suggestion
        message = MentionUtils.substituteUsernames(portalOwner, message);
```

File: component/core/src/test/java/org/exoplatform/social/core/utils/MentionUtilsTest.java [hunk 3]
```diff
@@ -33,15 +33,15 @@ public void testSubstituteUsernames(){
...
         assertEquals("hello <p><a href=\"http://test.com/portal/dw/profile/root\" target=\"_parent\">Root Root</a></p>", message);
     }
 
     @Test
     public void testWrongFormatMentionWithSubstituteUsernames(){
         String portalOwner = "dw";
         String message = "hello @ jhon";
-        message = MentionUtils.substituteUsernames(portalOwner, message);
+        message = MentionUtils.substituteUsernames(portalOwner, message, null);
```
Comment (boubaker) [inline component/core/src/test/java/org/exoplatform/social/core/utils/MentionUtilsTest.java]: Delete using null (to apply for all methods please)

```suggestion
        message = MentionUtils.substituteUsernames(portalOwner, message);
```

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue [hunk 4]
```diff
@@ -224,6 +224,12 @@ export default {
         // Thus, we destroy the ckEditor instance before setting new data.
         this.initCKEditorData(val || '');
       }
+    },
+    suggesterSpaceURL() {
+      if (this.suggesterSpaceURL) {
+        this.initCKEditor(true, this.value);
+      }
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: I think you miss an `else` as well to apply new suggestor options.

File: component/core/src/main/java/org/exoplatform/social/core/utils/MentionUtils.java [hunk 5]
```diff
@@ -39,7 +41,17 @@ public static String substituteUsernames(String portalOwner, String message) {
...
+                    SpaceService spaceService = CommonsUtils.getService(SpaceService.class);
+                    boolean isMember = spaceService.isMember(spaceService.getSpaceById(activity.getSpaceId()),identity.getRemoteId());
+                    if (isMember) {
+                        username = LinkProvider.getProfileLink(username, portalOwner);
+                    } else {
+                        username = identity.getProfile().getFullName();
+                    }
+                } else {
+                    username = LinkProvider.getProfileLink(username, portalOwner);
+                }
```
Comment (boubaker) [inline component/core/src/main/java/org/exoplatform/social/core/utils/MentionUtils.java]: No need to make the check here. In fact, the `MentionUtils` should be generic and doesn't change old activities when a member leaves a space by example.

```suggestion
```

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue [hunk 6]
```diff
@@ -224,6 +224,13 @@ export default {
         // Thus, we destroy the ckEditor instance before setting new data.
         this.initCKEditorData(val || '');
       }
+    },
+    suggesterSpaceURL() {
+      if (this.suggesterSpaceURL) {
+        this.initCKEditor(true, this.value);
+      } else {
+        this.initCKEditor(false, this.value);
+      }
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: Minor, but just for lisibility :smiley: 

```suggestion
      this.initCKEditor(!!this.suggesterSpaceURL, this.value);
```

**Review by boubaker — APPROVED** [review]

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue (hunk 6 above)
Comment (MayTekayaa) [reply webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: done!

**Pull request merged**